template/androidApp/src/main/java/io/bloco/template/features/list/ListScreen.kt

```diff
@@ -1,20 +1,61 @@
 package io.bloco.template.features.list
 
-import android.util.Log
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.gestures.Orientation
+import androidx.compose.foundation.gestures.scrollable
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.itemsIndexed
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.material.MaterialTheme
 import androidx.compose.material.Text
-import androidx.compose.material.TextField
 import androidx.compose.runtime.Composable
-import io.bloco.template.component.TemplateButton
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import com.google.accompanist.swiperefresh.SwipeRefresh
+import com.google.accompanist.swiperefresh.rememberSwipeRefreshState
+import io.bloco.template.R
+import io.bloco.template.component.Toast
 
 @Composable
 fun ListScreen(listViewModel: ListViewModel, openDetailsClicked: (String) -> Unit) {
-    /*TemplateButton(onClick = {
-        openDetailsClicked("1")
-    }) {
-        Text(text = "Open Details")
+    val bookListUpdateState by listViewModel.bookListUpdateState.collectAsState()
+    val books = listViewModel.books.collectAsState()
+
+    when (bookListUpdateState) {
+        ListViewModel.BookListUpdateState.ErrorFromAPI -> Toast(R.string.api_error)
```

Comment:
toast goes away, maybe it would be better a screen with a message, check wifi something like that. 
It would also be great make a retry action